src/eva/utilities/duration.py

```diff
@@ -0,0 +1,63 @@
+# (C) Copyright 2021-2022 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration. All Rights Reserved.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+# --------------------------------------------------------------------------------------------------
+
+
+from datetime import timedelta
+import re
+
+from eva.utilities.logger import Logger
+
+
+# --------------------------------------------------------------------------------------------------
+
+
+def iso_duration_to_timedelta(logger: Logger, iso_duration: str) -> timedelta:
+    """
+    Convert an ISO 8601 duration string to a datetime.timedelta object.
+
+    Parameters:
+    iso_duration (str): The ISO 8601 duration string.
+
+    Returns:
+    timedelta: The equivalent datetime.timedelta object.
+    """
+    pattern = re.compile(
+        r'P'  # starts with 'P'
+        r'(?:(?P<years>\d+)Y)?'  # years
+        r'(?:(?P<months>\d+)M)?'  # months
+        r'(?:(?P<days>\d+)D)?'  # days
+        r'(?:T'  # starts time part with 'T'
+        r'(?:(?P<hours>\d+)H)?'  # hours
+        r'(?:(?P<minutes>\d+)M)?'  # minutes
+        r'(?:(?P<seconds>\d+)S)?'  # seconds
+        r')?'  # end time part
+    )
+
```

Comment:
It might well be but it's not a standard library so I would prefer not to depend on it for something so simple. Not such an issue for GMAO but we might otherwise have to get the software approved for the operational machine and it's not worth it.